Userland/Games/BrickGame/BrickGame.cpp

```diff
@@ -0,0 +1,511 @@
+/*
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "BrickGame.h"
+#include <AK/Random.h>
+#include <LibConfig/Client.h>
+#include <LibGUI/MessageBox.h>
+#include <LibGUI/Painter.h>
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/Font/FontDatabase.h>
+#include <LibGfx/Point.h>
+
+using Position = Gfx::Point<int>;
+
+class Well final {
+    static constexpr size_t column_count = 10;
+    static constexpr size_t margin_left = 4;
+    static constexpr size_t margin_right = 32 - margin_left - column_count;
+    static constexpr size_t margin_bottom = 4;
+    static constexpr size_t row_count = 18 + margin_bottom;
+
+    static constexpr u32 empty_row = ~((~(~0u << column_count)) << margin_right);
```

Comment:
This definitely needs a comment to describe _why_ it represents an "empty row", cause otherwise it just looks like random operations to some constants.